oe_corporate_site_info.module

```diff
@@ -88,6 +108,59 @@ function oe_corporate_site_info_form_system_site_information_settings_alter(&$fo
   array_unshift($form['#submit'], '_oe_corporate_site_info_settings_form_submit');
 }
 
+/**
+ * Gets the URI without the 'internal:' or 'entity:' scheme.
+ *
+ * The following two forms of URIs are transformed:
+ * - 'entity:' URIs: to entity autocomplete ("label (entity id)") strings;
+ * - 'internal:' URIs: the scheme is stripped.
+ *
+ * This helper function duplicates the protected method
+ * LinkWidget::getUriAsDisplayableString().
+ *
+ * @param string $uri
+ *   The URI to get the displayable string for.
+ *
+ * @return string
+ *   The URI without the 'internal:' or 'entity:' scheme
+ *
+ * @see \Drupal\link\Plugin\Field\FieldWidget\LinkWidget::getUriAsDisplayableString()
+ */
+function _oe_corporate_site_info_get_privacy_default(string $uri): string {
+  $scheme = parse_url($uri, PHP_URL_SCHEME);
+  // By default, the displayable string is the URI.
+  $displayable_string = $uri;
+
+  // A different displayable string may be chosen in case of the 'internal:'
+  // or 'entity:' built-in schemes.
+  if ($scheme === 'internal') {
+    $uri_reference = explode(':', $uri, 2)[1];
+
+    // @todo '<front>' is valid input for BC reasons, may be removed by
+    //   https://www.drupal.org/node/2421941
+    $path = parse_url($uri, PHP_URL_PATH);
+    if ($path === '/') {
+      $uri_reference = '<front>' . substr($uri_reference, 1);
+    }
+
+    $displayable_string = $uri_reference;
+  }
+  elseif ($scheme === 'entity') {
+    [$entity_type, $entity_id] = explode('/', substr($uri, 7), 2);
+    // Show the 'entity:' URI as the entity autocomplete would.
+    // @todo Support entity types other than 'node'. Will be fixed in
+    //   https://www.drupal.org/node/2423093.
+    if ($entity_type == 'node' && $entity = \Drupal::entityTypeManager()->getStorage($entity_type)->load($entity_id)) {
```

Comment:
Load the entity outside the argument and check if it's an instance of NodeInterface.